portal_tables/sync_datasets.py

```diff
@@ -16,15 +16,35 @@ def add_missing_info(
         "".join(["[", d_id, "](", url, ")"]) if url else ""
         for d_id, url in zip(datasets["DatasetAlias"], datasets["DatasetUrl"])
     ]
+    url_pattern = re.compile(".*(synapse\.org).*")
+    alias_pattern = re.compile("^syn\d{7,8}$")
     datasets["grantName"] = ""
     datasets["themes"] = ""
     datasets["consortia"] = ""
     datasets["pub"] = ""
+    datasets["synapseLink"] = ""
     for _, row in datasets.iterrows():
+        is_in_synapse = None
+        for s in row["link"].split("]("):
+            s_match = re.match(url_pattern, s)
+            if s_match:
+                is_in_synapse = True
+        if is_in_synapse:
+            datasets.at[_, "synapseLink"] = datasets.at[_, "link"]
+        else:
+            alias_list = row["DatasetAlias"].split(",")
+            syn_link_list = []
+            for a in alias_list:
+                if re.match(alias_pattern, a):
+                    syn_link = "".join(["https://www.synapse.org/Synapse:", a])
+                    formatted_syn_link = "".join(["[", a, "](", syn_link, ")"])
+                    syn_link_list.append(formatted_syn_link)
```

Comment:
A list comprehension and f-string could be helpful here!

```suggestion
            syn_link_list = [
                f"[{a}](https://www.synapse.org/Synapse:{a})"
                for a in alias_list
                if re.match(alias_pattern, a)
            ]
```

Reply:
Also, not sure if you already knew this, but since you're using `re.match()` there's no need to include the `^` anchor in `alias_pattern`!  Since `re.match` looks for a pattern match at the start of the string already.  If you were using `re.search`, then the `^` would be necessary.

Reply:
Love the list comprehension, thanks for pointing that out!

> there's no need to include the ^ anchor in `alias_pattern`! Since `re.match` looks for a pattern match at the start of the string already. If you were using `re.search`, then the ^ would be necessary.

I think I started with `re.search` and didn't change the pattern, but that makes sense!

Reply:
It's always good to be explicit anyway, so I'd just keep the regex pattern as-is :-)

Just wanted to share as a general FYI!